html_telegraph_poster/utils.py

```diff
@@ -0,0 +1,53 @@
+# coding=utf8
+
+from urllib.parse import urlparse
+from .upload_images import upload_image
+from .converter import _fragments_from_string
+import lxml.html
+
+
+class DocumentPreprocessor:
+    def __init__(self, input_html):
+        self.input_html = input_html
+        self.parsed_document = self._parse_document()
+
+    def _parse_document(self):
+        fragments = _fragments_from_string(self.input_html)
+        document = fragments[0].xpath('/*')[0] if len(fragments) else None
+        return document
+
+    def get_processed_html(self):
+        return lxml.html.tostring(self.parsed_document, encoding='unicode')
+
+    def upload_all_images(self, base_url=None):
+        self._make_links_absoulte(base_url)
+        images = self.parsed_document.xpath('.//img[@src]')
+        for image in images:
+            old_image_url = image.attrib.get('src')
+            new_image_url = upload_image(old_image_url)
+            image.attrib.update({'src': new_image_url})
```

Comment:
How about ignoring exceptions of `upload_image()`? Or add a parameter to set `ignore_upload_failed`? Or other ideas about dealing with uploading exceptions?